drivers/adc/adc_b91.c

```diff
@@ -0,0 +1,369 @@
+		 ADC_GPIO_PD0, ADC_GPIO_PD1, ADC_VBAT)
+	}
+
+	return adc_pin;
+}
+
+/* Get ADC value */
+signed short adc_b91_get_code(void)
+{
+	signed short adc_code;
+
+	analog_write_reg8(areg_adc_data_sample_control,
+			  analog_read_reg8(areg_adc_data_sample_control) | FLD_NOT_SAMPLE_ADC_DATA);
+
+	adc_code = analog_read_reg16(areg_adc_misc_l);
+
+	analog_write_reg8(areg_adc_data_sample_control,
+		analog_read_reg8(areg_adc_data_sample_control) & (~FLD_NOT_SAMPLE_ADC_DATA));
+
+	if ((adc_code & BIT(SIGN_BIT_POSITION)) && !m_data.differential) {
+		adc_code = 0;
+	}
+
+	return adc_code;
+}
+
+/* ADC Context API implementation: start sampling */
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	m_data.repeat_buffer = m_data.buffer;
+
+	adc_power_on();
+
+	k_sem_give(&m_data.acq_sem);
+}
+
+/* ADC Context API implementation: buffer pointer */
+static void adc_context_update_buffer_pointer(struct adc_context *ctx, bool repeat_sampling)
+{
+	if (repeat_sampling) {
+		m_data.buffer = m_data.repeat_buffer;
+	}
+}
+
+/* Start ADC measurements */
+static int adc_b91_adc_start_read(const struct device *dev, const struct adc_sequence *sequence)
+{
+	int status;
+
+	/* Validate input parameters */
+	status = adc_b91_validate_sequence(sequence);
+	if (status != 0) {
+		return status;
+	}
+
+	/* Set resolution */
+	switch (sequence->resolution) {
+	case 14:
+		adc_set_resolution(ADC_RES14);
+		m_data.resolution_offset = 0;
+		break;
+	case 12:
+		adc_set_resolution(ADC_RES12);
+		m_data.resolution_offset = 2;
+		break;
+	case 10:
+		adc_set_resolution(ADC_RES10);
+		m_data.resolution_offset = 4;
+		break;
+	case 8:
+		adc_set_resolution(ADC_RES8);
+		m_data.resolution_offset = 6;
+		break;
+	default:
+		LOG_ERR("Selected ADC resolution is not supported.");
+		return -EINVAL;
+	}
+
+	/* Save buffer */
+	m_data.buffer = sequence->buffer;
+
+	/* Start ADC conversion */
+	adc_context_start_read(&m_data.ctx, sequence);
+
+	return adc_context_wait_for_completion(&m_data.ctx);
+}
+
+/* Main ADC Acquisition thread */
+static void adc_b91_acquisition_thread(const struct device *dev)
+{
+	int16_t adc_code;
+
+	while (true) {
+		/* Wait for Acquisition semaphore */
+		k_sem_take(&m_data.acq_sem, K_FOREVER);
+
+		/* Wait for ADC data ready */
+		while ((analog_read_reg8(AREG_ADC_DATA_STATUS) & ADC_DATA_READY)
+				!= ADC_DATA_READY) {
+		}
+
+		/* Perform read */
+		adc_code = (adc_b91_get_code() >> m_data.resolution_offset);
+		adc_code = adc_code << (m_data.differential ? 0 : 1);
+		*m_data.buffer++ = adc_code;
+
+		/* Power off ADC */
+		adc_power_off();
+
+		/* Release ADC context */
+		adc_context_on_sampling_done(&m_data.ctx, dev);
+	}
+}
+
+/* ADC Driver initialization */
+static int adc_b91_init(const struct device *dev)
+{
+	k_sem_init(&m_data.acq_sem, 0, 1);
+
+	k_thread_create(&m_data.thread, m_data.stack,
+			CONFIG_ADC_B91_ACQUISITION_THREAD_STACK_SIZE,
+			(k_thread_entry_t)adc_b91_acquisition_thread,
+			&m_data, NULL, NULL,
```

Comment:
`adc_b91_acquisition_thread` is defined to take `struct device *` as parameter.

Reply:
Fixed.